prediction_market_agent/agents/think_thoroughly_agent/think_thoroughly_agent.py

```diff
@@ -86,14 +90,28 @@ async def _arun(
 
 
 class CrewAIAgentSubquestions:
-    def __init__(self, model: str) -> None:
+    def __init__(self, model: str, memory: bool = True) -> None:
         self.model = model
         self.subgraph_handler = OmenSubgraphHandler()
         self.pinecone_handler = PineconeHandler()
+        self.memory = memory
+        if self.memory:
```

Comment:
```suggestion
```

Reply:
We only instantiate if memory=True, else do nothing.